dask/dataframe/__init__.py

```diff
@@ -17,6 +17,7 @@ def _dask_expr_enabled() -> bool:
             import dask_expr  # noqa: F401
         except ImportError:
             raise ValueError("Must install dask-expr to activate query planning.")
+        use_dask_expr = True
     return use_dask_expr
```

Comment:
I _think_ this is what we want, i.e., try to use `dask-expr` if enabled **and** `pandas>=2` is installed:

```suggestion
        return True
    return False
```

Reply:
Alternatively, you could also rewrite this to return early with 
```python
    use_dask_expr = dask.config.get("dataframe.query-planning")
    if use_dask_expr is False or use_dask_expr is None and Version(pd.__version__).major < 2:
        return False
    try:
        import dask_expr  # noqa: F401
    except ImportError:
        raise ValueError("Must install dask-expr to activate query planning.")
    return True
```

Reply:
I like the second version, added